contracts/BIFI/strategies/CompoundV2/StrategyCompoundV2Solidly.sol

```diff
@@ -0,0 +1,424 @@
+     * {borrowRateMax}       - A limit on how much we can push borrow risk.
+     * {borrowDepth}         - How many levels of leverage do we take.
+     * {minLeverage}         - The minimum amount of collateral required to leverage.
+     * {BORROW_DEPTH_MAX}    - A limit on how many steps we can leverage.
+     * {INTEREST_RATE_MODE}  - The type of borrow debt. Stable: 1, Variable: 2.
+     */
+    uint256 public borrowRate;
+    uint256 public borrowRateMax;
+    uint256 public borrowDepth;
+    uint256 public minLeverage;
+    uint256 constant public BORROW_DEPTH_MAX = 10;
+
+    /**
+     * @dev Helps to differentiate borrowed funds that shouldn't be used in functions like 'deposit()'
+     * as they're required to deleverage correctly.
+     */
+    uint256 public reserves;
+
+    uint256 public balanceOfPool;
+
+    /**
+     * @dev Events that the contract emits
+     */
+    event StratHarvest(address indexed harvester, uint256 wantHarvested, uint256 tvl);
+    event Deposit(uint256 tvl);
+    event Withdraw(uint256 tvl);
+    event StratRebalance(uint256 _borrowRate, uint256 _borrowDepth);
+    event ChargedFees(uint256 callFees, uint256 beefyFees, uint256 strategistFees);
+
+    constructor(
+        uint256 _borrowRate,
+        uint256 _borrowRateMax,
+        uint256 _borrowDepth,
+        uint256 _minLeverage,
+        ISolidlyRouter.Routes[] memory _outputToNativeRoute,
+        ISolidlyRouter.Routes[] memory _outputToWantRoute,
+        address[] memory _markets,
+        address _comptroller,
+        CommonAddresses memory _commonAddresses
+    ) StratFeeManager(_commonAddresses) {
+        borrowRate = _borrowRate;
+        borrowRateMax = _borrowRateMax;
+        borrowDepth = _borrowDepth;
+        minLeverage = _minLeverage;
+
+        iToken = _markets[0];
+        markets = _markets;
+        comptroller = _comptroller;
+        want = IVToken(iToken).underlying();
+
+        for (uint i; i < _outputToNativeRoute.length; ++i) {
+            outputToNativeRoute.push(_outputToNativeRoute[i]);
+        }
+
+        for (uint i; i < _outputToWantRoute.length; ++i) {
+            outputToWantRoute.push(_outputToWantRoute[i]);
+        }
+
+        output = outputToNativeRoute[0].from;
+        native = outputToNativeRoute[outputToNativeRoute.length -1].to;
+
+        _giveAllowances();
+        IComptroller(comptroller).enterMarkets(markets);
+    }
+
+    // puts the funds to work
+    function deposit() public whenNotPaused {
+        uint256 wantBal = availableWant();
+
+        if (wantBal > 0) {
+            _leverage(wantBal);
+            emit Deposit(balanceOf());
+        }
+    }
+
+    /**
+     * @dev Repeatedly supplies and borrows {want} following the configured {borrowRate} and {borrowDepth}
+     * @param _amount amount of {want} to leverage
+     */
+    function _leverage(uint256 _amount) internal {
+        if (_amount < minLeverage) { return; }
+
+        for (uint i = 0; i < borrowDepth; i++) {
```

Comment:
uint i is already established at 0 so no need to have it = 0. Also can do the same and move iterator to save gas.